Encounter "libtool_check_unique not found" error in apple_support@{1.18.0,1.18.1}

Encounter the following error after upgrading apple_support from 1.17.1 to 1.18.0:  https://github.com/secretflow/yacl/pull/464

```
libtool_check_unique (bazel-out/darwin_arm64-opt-exec-ST-d57f47055a04/bin/external/apple_support~~apple_cc_configure_extension~local_config_apple_cc/libtool_check_unique) not found. Please file an issue at https://github.com/bazelbuild/apple_support/issues
Archive of libgnu failed.
_____ END BUILD LOGS _____
```

#375 will fix this specific error. It won't fix an error in **rules_foreign_cc** where it assumes the compiler path is absolute, which results in this error:
```
-- Check for working CXX compiler: /private/var/tmp/_bazel_brentley.jones/6710248f6006564c68b8339361be6afc/sandbox/darwin-sandbox/6770/execroot/_main//bazel-out/darwin_arm64-opt-exec-ST-d57f47055a04/bin/external/apple_support~~apple_cc_configure_extension~local_config_apple_cc/wrapped_clang_pp - broken
CMake Error at /private/var/tmp/_bazel_brentley.jones/6710248f6006564c68b8339361be6afc/external/rules_foreign_cc~~tools~cmake-3.23.2-macos-universal/share/cmake-3.23/Modules/CMakeTestCXXCompiler.cmake:62 (message):
  The C++ compiler

    "/private/var/tmp/_bazel_brentley.jones/6710248f6006564c68b8339361be6afc/sandbox/darwin-sandbox/6770/execroot/_main//bazel-out/darwin_arm64-opt-exec-ST-d57f47055a04/bin/external/apple_support~~apple_cc_configure_extension~local_config_apple_cc/wrapped_clang_pp"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /private/var/tmp/_bazel_brentley.jones/6710248f6006564c68b8339361be6afc/sandbox/darwin-sandbox/6770/execroot/_main/bazel-out/darwin_arm64-opt/bin/external/mcl~/mcl-cmake.build_tmpdir/CMakeFiles/CMakeTmp
    
    Run Build Command(s):/private/var/tmp/_bazel_brentley.jones/6710248f6006564c68b8339361be6afc/sandbox/darwin-sandbox/6770/execroot/_main/external/rules_foreign_cc~~tools~ninja_1.12.1_mac_aarch64/ninja cmTC_d8ed1 && [1/2] Building CXX object CMakeFiles/cmTC_d8ed1.dir/testCXXCompiler.cxx.o
    [2/2] Linking CXX executable cmTC_d8ed1
    FAILED: cmTC_d8ed1 
    : && bazel-out/darwin_arm64-opt-exec-ST-d57f47055a04/bin/external/apple_support~~apple_cc_configure_extension~local_config_apple_cc/cc_wrapper.sh -D_FORTIFY_SOURCE=1 -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -g0 -O2 -DNDEBUG -std=c++14 -DNS_BLOCK_ASSERTIONS=1 -fdebug-prefix-map=__BAZEL_EXECUTION_ROOT__=. -fdebug-prefix-map=__BAZEL_XCODE_DEVELOPER_DIR__=/PLACEHOLDER_DEVELOPER_DIR -isysroot __BAZEL_XCODE_SDKROOT__ -F__BAZEL_XCODE_SDKROOT__/System/Library/Frameworks -F__BAZEL_XCODE_DEVELOPER_DIR__/Platforms/MacOSX.platform/Developer/Library/Frameworks -no-canonical-prefixes -pthread -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\" -target arm64-apple-macosx13.0 -fdiagnostics-color=always -Wno-unused-command-line-argument -std=c++17  -arch arm64 -isysroot /Applications/Xcode-16.1.0.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk -mmacosx-version-min=13.0 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -lc++ -Wl,-oso_prefix,__BAZEL_EXECUTION_ROOT_NO_SANDBOX__/ -headerpad_max_install_names -no-canonical-prefixes -target arm64-apple-macosx13.0 -fobjc-link-runtime -dead_strip -Xlinker -Os -framework Foundation -Wl,-no_warn_duplicate_libraries -lm CMakeFiles/cmTC_d8ed1.dir/testCXXCompiler.cxx.o -o cmTC_d8ed1   && :
    /bin/sh: bazel-out/darwin_arm64-opt-exec-ST-d57f47055a04/bin/external/apple_support~~apple_cc_configure_extension~local_config_apple_cc/cc_wrapper.sh: No such file or directory
```

It needs to get an absolute path to `cc_wrapper.sh` since it's calling it from outside of Bazel's execution root. This breakage is from us changing to use `action_config(tools = [tool(tool = ... )])` from `action_config(tools = [tool(path = ... )])` (which would produce an absolute path it seems).

--- a/crosstool/libtool.sh.tpl
+++ b/crosstool/libtool.sh.tpl
@@ -23,7 +23,15 @@
 
 set -eu
 
-readonly xcrunwrapper="%{xcrunwrapper}"
+parent_dir="$(cd "${BASH_SOURCE%/*}" && pwd)"
+readonly libtool_check_unique_relative="%{libtool_check_unique}"
+
+# `libtool_check_unique`` is generated and placed next to this script, so we can
+# use it's parent directory, which is relative to the execution root, to find
+# the absolute path to the execution root
+readonly execution_root="${parent_dir%/"${libtool_check_unique_relative%/*}"}"
+
+readonly xcrunwrapper="$execution_root/%{xcrunwrapper}"
 function invoke_libtool() {
   # Just invoke libtool via xcrunwrapper
   "$xcrunwrapper" libtool "$@" \
@@ -33,7 +41,7 @@ function invoke_libtool() {
   # ...and not silencable via a flag.
 }
 
-readonly libtool_check_unique="%{libtool_check_unique}"
+readonly libtool_check_unique="$execution_root/$libtool_check_unique_relative"
 if [ ! -f "$libtool_check_unique" ] ; then
   echo >&2 "libtool_check_unique ($libtool_check_unique) not found. Please file an issue at https://github.com/bazelbuild/apple_support/issues"
   exit 1
@@ -71,7 +79,7 @@ ARGS=()
 handle_filelist=0
 keep_next=0
 
-readonly make_hashed_objlist="%{make_hashed_objlist}"
+readonly make_hashed_objlist="$execution_root/%{make_hashed_objlist}"
 function parse_option() {
   local -r ARG="$1"
   if [[ "$handle_filelist" == "1" ]]; then

